interfaces/acados_template/acados_template/mpc_utils.py

```diff
@@ -117,3 +118,198 @@ def create_model_with_cost_state(ocp: AcadosOcp) -> Tuple[AcadosModel, np.ndarra
     model.f_impl_expr = ca.vertcat(model.f_impl_expr, cost_state_dot-cost_dot)
 
     return model, ocp.parameter_values
+
+
+def detect_constr(model: AcadosModel, constraints: AcadosOcpConstraints, stage_type: str):
+    """
+    - stage_type: allowed values "initial", "path", "terminal"
+    """
+    x = model.x
+    u = model.u
+    z = model.z
+    nx = x.shape[0]
+    nu = u.shape[0]
+    nz = z.shape[0]
+
+    if isinstance(x, ca.SX):
+        isSX = True
+    else:
+        raise ValueError('Constraint detection only works for casadi.SX!')
```

Comment:
what is missing to have this work with MX?

Reply:
Not sure, I was copying/pasting from the matlab files.
I will test if also MX are supported.